chapter_2/3_dbscan_variations.tex

```diff
@@ -0,0 +1,132 @@
+
+\begin{figure}[h]
+\centering
+\includegraphics[scale=0.27]{chapter_2/files/dbscan_plus_comparison.png}
+\caption{A comparison of DBSCAN++ sampling methods with traditional DBSCAN clustering. Performance is fairly consistent across the three examples, with clusters discovered using K-Centers sampling perhaps slightly closer to the original than uniform sampling. Source: \cite{dbscanpp}}
+\label{fig:dbscan_plus}
+\end{figure}
+
+\noindent \textbf{OPTICS}
+
+An important issue with DBSCAN is its inability to capture clusters of varying density. Furthermore, cluster density must be manually tuned with the $\epsilon$ bandwidth parameter. Small changes in $\epsilon$ may devastate the quality of the resulting clusters.
+
+A popular solution to both problems is OPTICS: Ordering Points To Identify the Clustering Structure. The basic setup is similar to DBSCAN, but defines two additional metrics: core distance and reachability distance. Suppose $p_{(m)}$ is the $m$th closest point to $p$. Then with $\epsilon$ bandwidth and $m$ minimum points to qualify a cluster, we define:\cite{wiki:optics}, \cite{medium:optics}
+
+\begin{align}
+    \text{core-dist}_{\epsilon, m}(p) &= 
+        \begin{cases}
+            d(p, p_{(m)}) & |N_\epsilon(p)| \geq m \\
+            \text{undefined} & \text{otherwise} \\
+        \end{cases}\\
+    \text{reachability-dist}_{\epsilon, m}(p, o) &= 
+        \begin{cases}
+            \max \lbrace d(p, o), \text{core-dist}_{\epsilon, m} (p) \rbrace & |N_\epsilon(p)| \geq m \\
+            \text{undefined} & \text{otherwise} \\
+        \end{cases}\\
+\end{align}
+
+First, observe that both metrics apply only to core points. For border and noise points, they are undefined. Indeed, under most OPTICS implementations, border and noise points do not earn membership in any cluster, and are ignored.
+
+\begin{figure}[h]
+\centering
+\includegraphics[scale=0.42]{chapter_2/files/optics_distances.png}
+\caption{An example of core and reachability distances, from a core point $p$ and $m=4$. Source: \cite{medium:optics}}
+\label{fig:dbscan_plus}
+\end{figure}
+
+Core distance is a measure of a core point's local density. If $\text{core-dist}_{\epsilon, m}(p) = \epsilon'$, then $\epsilon' << \epsilon$ implies that the core point must be in a region of high density.
+
+Reachability distance is a measure of the distance between a core point $p$ and a nearby point $o$. In most implementations, $o$ is drawn from $N_\epsilon(p)$, in which case reachability distancec will always be less than $\epsilon$. If $o$ is especially close to $p$, then reachability distance is the same as core distance. Otherwise, it is the actual distance between the two points.
+
+The goal of OPTICS is to assign a reachability distance to all core points between themselves and the closest, untraversed neighbor. In this way, the algorithm proceeds along a greedy shortest path traversal of the dataset, generating a reachability plot that encodes the clusters.
```

Comment:
It refers to a neighbor for which a reachability distance has not been assigned. I've removed the "untraversed" part since it doesn't matter for building intuition, but the specifics are encoded in the pseudocode.